src/App.js

```diff
@@ -4,16 +4,43 @@ import House from './components/House.js';
 import {Howler} from 'howler';
 
 class App extends Component {
-  render() {
-    return (
-      <div className="App">
-        <House/>
-      </div>
-    );
+  constructor(props) {
+        super(props);
+        this.state = {gamestate: 'game'};
+        
   }
+  
+  // ether bind or use es6
+  changeGameState = (newstate) => {
+      console.assert(newstate === 'game' || newstate === 'menu',
+       "Cannot set game state invalid arg input.");
```

Comment:
I would usually put the bad input in the error message to make it easier to debug

Reply:
This is true, it does help. Unless the error is the end of string null terminator missing.

Reply:
I also want to first check the security risk, if any. I want to be very careful with variables when I don't know what they contain. Have you heard of that sql hack that programmer did a long time ago where the user's input was restructured in an sql command for searching. He was able to insert his own request and have the whole table output.

Reply:
I'm familiar with sql injection attacks, and good on you for watching out for injection vulnerabilities, but I think in this case we're not so worried. We don't usually hit the null termination issue in js, that's more of a c thing.